### PR TITLE
Adding JX examples

### PR DESCRIPTION
**Pull request by AndrewLitteken**
Adding the JX examples for blast, shrimp, hecil and ssaha.  Added a new rule to bwa.jx that corrected input and output errors.

### CODE REVIEW

**Comment by dthain** [comment]
@nhazekam please take a look at this

File: shrimp/context.jx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+	"QUERY": "query.csfasta",
+	"GENOME": "genome.fasta",
+
+	"MAPPER": "./rmapper-cs",
+	"MOPTIONS": "-M fast -M 50bp",
+	
+	"SPLIT": "./split_fasta",
+	"SPLITS": 10000,
+	"SPLIT_SIZE": 762,
```
Comment (nhazekam) [inline shrimp/context.jx]: As with above add TOTAL_SEQ and remove SPLIT_SIZE.

File: ssaha/context.jx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	"QUERY": "query.fastq",
+	
+	"DB": "db.fastq",
+
+	"SSAHA": "./ssaha2",
+	"OPTIONS": "-tags 1 -rtype abi -score 30 -seeds 5 -output ssaha2",
+
+	"SPLIT": "./fastq_split.pl",
+	"SPLITS": 1000,
+	"SPLIT_SIZE": 100,
```
Comment (nhazekam) [inline ssaha/context.jx]: Same as above.

File: json2jx-doc.md [hunk 3]
```diff
@@ -0,0 +1,889 @@
+# Converting JSON to JX #
```
Comment (nhazekam) [inline json2jx-doc.md]: @trshaffer Please take a look. I perused this file, but this is more your domain.

Comment (trshaffer) [reply json2jx-doc.md]: I think this looks good. It's helpful as an intro for workflow people who don't necessarily care about all the details of JX

File: json2jx-doc.md [hunk 4]
```diff
@@ -0,0 +1,889 @@
...
+"command": format(
+	"cat %s > Corrected_ref.fasta",
+	join([format("Corrected_ref.fasta.%d", x) for x in r    ange(SPLIT_SIZE)])
+)
+```
+When expanded, this creates an analagous string to be executed.
+
+## JSON to JX ##
+
+### JSON Format ###
```
[the remark this answers is not in the document]
Comment (trshaffer) [reply json2jx-doc.md]: Agree, this could just point to the corresponding files

**Comment by dthain** [comment]
@AndrewLitteken please address the points from Nick and Tim, and then we can merge it.

**Pull request merged**